File: Python3/프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    result = 0
    for i in range(left, right + 1):
        if count_divisor(i) % 2 == 0:
            result += i
        else:
            result -= i
    
    return result

def count_divisor(num):
    if num <= 0:
        return 0
    
    divisor_count = 1
    i = 2
    
    while i * i <= num:
        count = 0
        while num % i == 0:
            count += 1
            num //= i
        
        divisor_count *= (count + 1)
        i += 1
        
    if num > 1:
        divisor_count *= 2
    
    return divisor_count
            

File: Swift/프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.swift
import Foundation

func solution(_ left:Int, _ right:Int) -> Int {
    var result = 0 
    for i in left...right {
        let count = numberOfDivisors(i)
        
        if count % 2 == 0 {
            result += i
        } else {
            result -= i
        }
    }
    
    return result
}

func numberOfDivisors(_ n: Int) -> Int {
    if n <= 0 {
        return 0
    }
    
    var num = n
    var divisorCount = 1
    var i = 2
    
    while i * i <= num {
        var count = 0
        while num % i == 0 {
            count += 1
            num /= i
        }
        divisorCount *= (count + 1)
        i += 1
    }
    
    if num > 1 {
        divisorCount *= 2
    }
    
    return divisorCount
}
